### DISCUSSION TITLE
Intro

### DISCUSSION THREAD

**Discussion by Fraganya** [Fraganya/weakest-link#3] [Announcements]
Welcome to the weakest link repo.

**Comment by Fraganya** [comment]
@todo:

1. Review existing codebase
2. Develop documentation for game rules
3. Design - UI Mockups, Service components, DB
4. Development iterations  
- Coding 
- Tests



Tech Stack:
- Typescript
- Next.JS  (React)
- Tailwind
- Node  => open to  Go, Java, Python
- Message broker  :  MQTT | Mosquito , RabbitMQ, Redis , Apache Kafka, NATS.IO

**Reply by bdsaved** [reply]
I am thinking we stick to node and keep the architecture Typescript strickly.

**Comment by Fraganya** [comment]
Possibly use Doxygen for documentation

**Comment by bdsaved** [comment]
Maybe Also include an android application too

**Reply by Fraganya** [reply]
Great idea, ndekti the backend API has to be designed in a way to be usable by a web front end and a mobile app

**Reply by bdsaved** [reply]
Yes, so I am thinking maybe having the backend API as an event driven API, building it with websockets or messaging but not restful. Or maybe the normal operations would be RESTful and then the game logic could be delivered differently.

**Reply by Fraganya** [reply]
yea yea , that could work, for the basic stuff, RESTful APIs would be ok. for the other stuff we could use websockets.

**Comment by Fraganya** [comment]
The weakest link is a real world television game show.In this game a group of contestants (players) work together to try and win as much money as possible by answering a series of questions. The number of contestants range from 5-9 (varies among versions).If a person answers the question correctly, the money chain moves to the next stage otherwise if wrong the money chain restarts.A contestants has the option of banking the money before answering the question but the chain restarts.If 9 questions are answered correctly and no one banks the money is automatically added to the jackpot and the round stops.The time allowed for each round decreases as the round increases. If the contestant takes too long to answer, it is considered to be incorrect and the answer is given. At the end of each round all contestants are allowed to vote for who they think is the weakest link (The person dragging them down) in then just ended round.The one with the most votes is eliminated and if there is a tie the strongest link (The person who had the most correct answers or banked the most money) gets to chose who leaves among the ties.The strongest link gets to start in the next round and if was voted of the second strongest link gets to go first.In the final round an allowance is made to generate more money and there is no voting. This is called the triple stake round which usually spans 90 seconds.After that comes the head-to-head contest in which each of the two remaining contestants battle it out with 5 questions asked to each contestant.if there is a tie the game continues to a sudden death.The winner gets all the money whilst the loser goes home with nothing.Hope this information was useful.
